apps/commercetools/src/api/products.ts

```diff
@@ -19,42 +21,71 @@ function productTransformer({ projectKey, locale }: ConfigurationParameters) {
   };
 }
 
-export async function fetchProductPreviews(
+async function fetchAllProductPreviews(
+  client: ReturnType<typeof createClient>,
+  products: ProductProjection[] = [],
   skus: string[],
-  config: ConfigurationParameters
-): Promise<Product[]> {
+  offset: number = 0
+): Promise<ProductProjection[]> {
   if (skus.length === 0) {
     return [];
   }
 
-  const client = createClient(config);
+  let nextPageProducts: ProductProjection[] = [];
+
   const response = await client
     .productProjections()
     .search()
     .get({
       queryArgs: {
         'filter.query': [`variants.sku:${skus.map((sku) => `"${sku}"`).join(',')}`],
+        offset,
+        limit: FETCH_STEP,
       },
     })
     .execute();
 
   if (response.statusCode === 200) {
-    const products = response.body.results.map(productTransformer(config));
-    const foundSKUs = products.map((product: Product) => product.sku);
-    const missingProducts = skus
-      .filter((sku) => !foundSKUs.includes(sku))
-      .map((sku) => ({
-        sku,
-        image: '',
-        id: '',
-        name: '',
-        isMissing: true,
-      }));
-    return [...products, ...missingProducts];
+    const hasNextPage = response.body.offset + response.body.count < response.body.total!;
+    const combinedResults = [...products, ...response.body.results];
+
+    if (hasNextPage) {
+      nextPageProducts = await fetchAllProductPreviews(
```

Comment:
FWIW: this will lead to sequential requests. I would see if you can parallelise them as it would technically only take ~1s for 1000 items (assuming 500ms/req) while this sequential fetching would take ~10s